pkg/scheduler/dependency_graph.go

```diff
@@ -434,37 +482,74 @@ func (sched *Scheduler) BuildDependencyGraph(options interfaces.DependencyGraphO
 	depGraph := NewDependencyGraph(sched, options)
 	rootContext := &GraphContext{scheduler: sched, graph: depGraph, flow: flow, args: options.Args}
 
-	if _, ok := dependencies[fullFlowName]; !ok {
-		log.Printf("Flow %s is empty", options.FlowName)
-		return depGraph, nil
+	replicas, deleteReplicas, err := sched.allocateReplicas(flow, options)
+	if err != nil {
+		return nil, err
 	}
 
-	type Block struct {
-		dependency        client.Dependency
-		scheduledResource *ScheduledResource
-		parentContext     *GraphContext
+	err = sched.fillDependencyGraph(rootContext, resDefs, dependencies, flow, replicas, false)
+	if err != nil {
+		return nil, err
 	}
-
-	replicas, err := sched.allocateReplicas(flow, options)
+	err = sched.fillDependencyGraph(rootContext, resDefs, dependencies, flow, deleteReplicas, true)
 	if err != nil {
 		return nil, err
 	}
 
+	for _, value := range depGraph.graph {
+		value.RequiredBy = unique(value.RequiredBy)
+		value.Requires = unique(value.Requires)
+		value.usedInReplicas = unique(value.usedInReplicas)
+	}
+
+	if len(deleteReplicas) > 0 {
+		optionsCopy := options
+		optionsCopy.ReplicaCount = 0
+		optionsCopy.MinReplicaCount = 0
+		optionsCopy.FixedNumberOfReplicas = false
+		optionsCopy.Silent = true
+
+		allReplicas, _, err := sched.allocateReplicas(flow, optionsCopy)
```

Comment:
with the options provided it won't allocate new replicas, just return the list of existing replicas.